### PR TITLE
Use EasyCodingStandard over phpcs, php_codesniffer and phpmd

### LINKED ISSUE #7: Make use of slevomat and object-calisthenics to move from phpmd
These rules of `phpmd`:

- `ShortVariable`
- `ShortMethodName`
- `UnusedPrivateMethod`
- `ExcessivePublicCount`
- `ExcessiveClassLength`
- `ExcessiveMethodLength`

**are already implemented in these 2 packages** for PHP_CodeSniffer as Sniffs:

- https://github.com/object-calisthenics/phpcs-calisthenics-rules - you gonna love it, if you like strictness, and I can see you do :)
- https://github.com/slevomat/coding-standard

And few more useful rules, that could be added to `rulesetCS.xml`. Both packages are well maintained and up-to-date with PHP_CodeSniffer 3.x.

By using them, you could easily drop `phpmd` that never fix the code for you and move to `PHP_CodeSniffer` and `PHP CS Fixer` who do.

### PR DESCRIPTION
**Pull request by TomasVotruba**
**Description, reason for the PR:** With bumping to PHP 7.1 in #11, having 3 tools is not neccesary.

- [Using own rules](https://github.com/shopsys/coding-standards/blob/c602722e4ee701d95f1d2abc3a57e3d732a1ab01/rulesetCS.xml#L64) requires studying PHP_CodeSniffer internals, which is painful and completely unneccessary
- 1 tools is better than 3: to learn, understand, debug and extend

**New feature:** No

**BC breaks:** No

**Fixes issues:** Closes #7 

**Licence:** MIT

### CODE REVIEW

**Comment by TomasVotruba** [comment]
@vitek-rostislav I've integrated notes from our personal review. It's ready to merge :+1: 

I'll handle PHPMD migration in next PR

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,9 +7,13 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
...
 
 ## [Unreleased]
+
+### Added
+- [#12] Added [EasyCodingStandard](https://github.com/Symplify/EasyCodingStandard)
+
 ### Changed
 - `OrmJoinColumnRequireNullableFixer` marked as *risky* (@sustmi)
-- dropped support of PHP 7.0 (@vitek-rostislav)
+- [#11] dropped support of PHP 7.0 (@vitek-rostislav)
```
Comment (TomasVotruba) [inline CHANGELOG.md]: @todo Show https://github.com/Symplify/ChangelogLinker to @vitek-rostislav

File: easy-coding-standard.neon [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+    - ShopSys\CodingStandards\Sniffs\ForbiddenExitSniff
+    - ShopSys\CodingStandards\Sniffs\ForbiddenSuperGlobalSniff
+    - ShopSys\CodingStandards\CsFixer\MissingButtonTypeFixer
+    - ShopSys\CodingStandards\CsFixer\OrmJoinColumnRequireNullableFixer
+
+    - PHP_CodeSniffer\Standards\PSR1\Sniffs\Methods\CamelCapsMethodNameSniff
+
+    # Fine-tuning of PSR-2 standard
+    # Sometimes, there is no other way than have longer lines. For example when using fully-qualified namespaces in type hinting comments.
+    PHP_CodeSniffer\Standards\Generic\Sniffs\Files\LineLengthSniff:
```
Comment (vitek-rostislav) [inline easy-coding-standard.neon]: I am still not sure about this sniff, the psr-2 says: "There MUST NOT be a hard limit on line length". I think the best solution would be to omit the rule completely.

**Review by vitek-rostislav — CHANGES_REQUESTED** [review]
Hi, thanks for the PR, the only problem I have is with the line length rule.

File: easy-coding-standard.neon (hunk 2 above)
Comment (TomasVotruba) [reply easy-coding-standard.neon]: Absolutely fine with me. 

I'm slowly moving to "automatically [fix only valid cases](https://github.com/Symplify/CodingStandard#arguments-should-be-on-the-samestandalone-line-to-fit-line-length) and do not annoy user about it" approach

I'll handle that in a minute

**Comment by TomasVotruba** [comment]
Done :+1:

**Review by vitek-rostislav — APPROVED** [review]

**Pull request merged**